src/reporters/build_kite_cli.ts

```diff
@@ -0,0 +1,187 @@
+    this.recordVideo = options.recordVideo;
+    this.screenshotsPath = join(CACHE_PATH, 'screenshots');
+  }
+
+  override onJourneyStart(journey: Journey, {}: JourneyStartResult) {
+    this.write(`\n--- Journey: ${journey.name}`);
+  }
+
+  override onStepEnd(journey: Journey, step: Step, result: StepEndResult) {
+    const { status, end, start, error } = result;
+    const message = `${symbols[status]}  Step: '${
+      step.name
+    }' ${status} (${renderDuration((end - start) * 1000)} ms)`;
+    this.write(indent(message));
+    if (error) {
+      this.write(renderError(error));
+    }
+    this.metrics[status]++;
+    if (!this.journeys.has(journey.name)) {
+      this.journeys.set(journey.name, []);
+    }
+    this.journeys.get(journey.name)?.push({ name: step.name, ...result });
+  }
+
+  override async onJourneyEnd(
+    journey: Journey,
+    { error, start, end, status, options }: JourneyEndResult
+  ) {
+    const { screenshots, outputDir } = options;
+    const { failed, succeeded, skipped } = this.metrics;
+    const total = failed + succeeded + skipped;
+    if (total === 0 && error) {
+      this.write(renderError(error));
+    }
+    const message = `${symbols[status]} Took  (${renderDuration(
+      end - start
+    )} seconds)`;
+    this.write(message);
+
+    if (outputDir) {
+      const FAILED_STEPS_PATH = join(outputDir, 'failed_steps');
+      await fs.promises.mkdir(FAILED_STEPS_PATH, { recursive: true });
+
+      await writeScreenshotsToPath(screenshots, outputDir, status);
+    }
+  }
+
+  override onEnd() {
+    const failedJourneys = Array.from(this.journeys.entries()).filter(
+      ([, steps]) => steps.some(step => step.status === 'failed')
+    );
+
+    if (failedJourneys.length > 0) {
+      failedJourneys.forEach(([journeyName, steps]) => {
+        const name = red(`Journey: ${journeyName} 🥵`);
+        this.write(`\n+++ ${name}`);
+        steps.forEach(stepResult => {
+          const { status, end, start, error, name: stepName } = stepResult;
+          const message = `${
+            symbols[status]
+          }  Step: '${stepName}' ${status} (${renderDuration(
+            (end - start) * 1000
+          )} ms)`;
+          this.write(indent(message));
+          if (error) {
+            this.write(renderError(error));
+          }
+        });
+      });
+    }
+
+    const successfulJourneys = Array.from(this.journeys.entries()).filter(
+      ([, steps]) => steps.every(step => step.status === 'succeeded')
+    );
+
+    const dirPath = this.recordVideo?.dir;
+    if (dirPath) {
+      successfulJourneys.forEach(([journeyName]) => {
+        const filePath = join(dirPath, journeyName + '.webm');
+        try {
+          fs.unlinkSync(filePath);
+        } catch (e) {
+          // eslint-disable-next-line no-console
+          console.log('Failed to delete video file for path ' + filePath);
+        }
+      });
+    }
+
+    const { failed, succeeded, skipped } = this.metrics;
```

Comment:
Better to move this inside the base reporter and print a summaru/epilogue of the tests and you can invoke the base reporter since all reporters extend it.

Reply:
Right now we repeat a lot of the same stuff from base reporter.